src/main/java/uk/gov/hmcts/reform/waworkflowapi/clients/service/ExternalTaskWorker.java

```diff
@@ -48,20 +52,49 @@ public void setupClient() {
     }
 
     public void checkIdempotency(ExternalTask externalTask, ExternalTaskService externalTaskService) {
-        Boolean isDuplicate =  externalTask.getVariable("isDuplicate");
-        if (isDuplicate == null || isDuplicate) {
-            Map<String, Object> processVariables = singletonMap(
-                "isDuplicate",
-                false
-            );
+        IdempotentId idempotentId = getIdempotentId(externalTask);
 
-            LOGGER.info("Duplicate was hit.");
+        Optional<IdempotentKeys> idempotentRow = idempotentKeysRepository.findById(idempotentId);
```

Comment:
How are database exceptions handled?  So for instance if there's a lock exception or a connectivity exception. What goes back to Camunda?  Maybe we can have quick chat about that scenario.

Reply:
The ticket does not mention error handling scenarios, so I did not think about that much. But I thought to bring this up with you, yes.

Reply:
Yes, let's talk about this and add to the unhappy paths ticket.